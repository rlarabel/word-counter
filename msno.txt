MissingNo. (Japanese: Ketsuban), short for "Missing Number" and sometimes spelled without the period, is an unofficial Pokemon species found in the video games Pokemon Red and Blue. Due to the programming of certain in-game events, players can encounter MissingNo. via a glitch.
Encountering MissingNo. causes graphical anomalies and changes gameplay by increasing the number of items in the sixth entry of the player's inventory by 128. This beneficial effect resulted in the glitch's coverage by strategy guides and game magazines, while game publisher Nintendo warned that encountering the glitch may corrupt players' game data. IGN noted MissingNo.'s appearance in Pokemon Red and Blue was one of the most famous video game glitches and commented on its role in increasing the series' popularity. Fans have attempted to rationalize and incorporate MissingNo. as part of the games' canon as an actual in-game character, and sociologists have studied its impact on both players and gaming culture as a whole.


== History ==
Developed by Game Freak and published by Nintendo, the Pokemon series began in Japan in 1996 with the release of the Pokemon Red and Blue video games for the Game Boy. In these games, the player assumes the role of a Pokemon Trainer whose goal is to capture and train creatures called Pokemon. Players use the creatures' special abilities to combat other Pokemon, and certain abilities also grant new ways to navigate the game's world, such as instantaneous travel between two areas. The ultimate goal of the games is to complete the entries in the Pokemon index (Pokedex), a comprehensive Pokemon encyclopedia, by capturing, evolving, and trading to obtain creatures from all 151 Pokemon species.MissingNo. is not one of the official Pokemon species that players are meant to encounter, but it is accessible to players of European and North American copies of the games by a glitch. Nintendo of America first documented the events that cause MissingNo. to appear in the May 1999 issue of Nintendo Power. The company warned that "any contact with it (even if you don't catch it) could easily erase your game file or disrupt your graphics". Game developers did not remove several glitches from the 2016 re-releases of Pokemon Red and Blue on the Nintendo 3DS Virtual Console, and players can encounter MissingNo. in these versions of the games.


== Characteristics ==

A player can encounter a MissingNo. in Pokemon Red and Blue by following a series of steps. First, the player watches an in-game tutorial for Pokemon capture in the game's Viridian City location. Second, the player uses a Pokemon with the "Fly" move to instantly travel to the game's Cinnabar Island location. Finally, the player uses a Pokemon with the "Surf" move to travel up and down the eastern shore of the island until a MissingNo. appears.These events manipulate the game's random encounter system to generate a Pokemon with an invalid identifier. Each area within the game assigns values to a data buffer to represent the Pokemon that can be encountered in that area. However, some areas-such as Cinnabar Island-do not overwrite the data in this buffer, so the data from the previous area is used instead. During the Viridian City in-game tutorial, the player character's name is temporarily overridden to read "Old Man", and the player character's actual name is temporarily copied to that same data buffer. If the player travels directly to Cinnabar Island after viewing this tutorial, the player's name will be read as the Pokemon that can be randomly encountered in that area. Due to the player character's name not being intended to be read as this kind of data, the game can attempt to generate an encounter with a Pokemon with an invalid identifier, such as MissingNo. Fans have dubbed this method of encountering MissingNo. the "old man glitch".As with any wild Pokemon, players may flee from, fight, or capture MissingNo. After an encounter with MissingNo., the quantity of the sixth item in the player's inventory is increased by 128, and the game's Hall of Fame Pokemon gallery becomes glitched. Temporary graphical glitches may also occur, which can be removed by viewing the statistics page for another non-glitched Pokemon or resetting the console.A captured MissingNo. is functional as a Pokemon and appears in the games' Pokedex as number 000. The games classify it as a hybrid Bird/Normal-type Pokemon even though the category of Bird-type Pokemon was cut from the games before release. It commonly appears with a scrambled block-like form commonly described as a "backward L-shape", but depending on the player character's name, it can also appear as one of three ghost or fossil sprites not used by other Pokemon.


== Reaction and reception ==
Calling MissingNo. a "programming quirk", Nintendo warned against encountering it, saying players could possibly have to restart the game from the beginning to remove the graphical glitches. Despite Nintendo's warning, information on how to encounter MissingNo. was printed in several magazines and player's guides due to its perceived positive effect. Certain players attempted to sell tips on capturing MissingNo. for up to $200.Despite it not being an intentional part of the game, in 2009, IGN included MissingNo. in its list of the top video game Easter eggs, citing its usefulness in replicating the game's rarer items and in a later article, calling it an "unforgettable" glitch that helped push the original games to "gaming super stardom". The book 100 Greatest Video Game Characters describes MissingNo. as an example of a player-produced character and counterplay, adding, "as a cultural artifact, MissingNo[.] celebrates the creativity of deviant and subversive forms of play". Some fans have even created costumes based on MissingNo.A 2007 paper, co-authored by sociologist William Sims Bainbridge and his daughter Wilma Bainbridge, then a cognitive science student, refers to MissingNo. as "one of the most popular glitches ever in game history". In a 2019 comment provided to Ars Technica, Wilma Bainbridge, then a post-doctoral fellow at the National Institute of Mental Health, expressed her belief that MissingNo. helped contribute to the widespread interest in glitch hunting and speedrunning in video games. Alternatively in the book Pikachu's Global Adventure: The Rise and Fall of Pokemon, professor of education Julian Sefton-Green noticed that in his study of his son's reaction to MissingNo.'s usage as a cheat, the child's outlook towards the game was altered drastically, and added that the presence of such elements as a result broke the illusion of the game as an enclosed world and reminded him that "at heart, it is a computer program".The 2008 book Playing with Videogames discusses how players compare notes on MissingNo. and to assess and critique each other's findings. James Newman, the author, commented that unusually MissingNo. caused players to celebrate the game's imperfections. In a 2017 article, University of Portsmouth professor Lincoln Geraghty examined fan theories that the glitch was a cut Pokemon related to the Pokemon Kangaskhan and Cubone, describing the theory as "an established work of fanon [fan canon]". He further elaborated in a 2019 Ars Technica article that "fans' desire to incorporate MissingNo[.] into the Pokemon world stems partly from a hope that the glitch may have been deliberate", as well as to compensate for the glitch's lack of backstory within Pokemon canon. In the same article, Newman argued that the existence of such fan theories "reveals a belief in the fundamental reality of Pokemon as entities that are given an opportunity to show themselves through the game, rather than being constructed out of code".


== References ==


== External links ==

 Media related to MissingNo. at Wikimedia Commons
MissingNo. on Bulbapedia
MissingNo.'s Glitchy Appearance Explained -  Retro Game Mechanics Explained
Why MissingNo Multiplies Items - Live Overflow
