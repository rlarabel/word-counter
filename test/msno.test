echo "Running tests..."
echo

make
./wc msno.txt > TESTOUT

if [ $? -eq 0 ] ; then
  echo "Pass: Program exited zero"
else
  echo "Fail: Program did not exit zero"
  rm TESTOUT
  exit 1
fi

diff test/msno.out TESTOUT > DIFFOUT

if [ $? -eq 0 ] ; then
  echo "Pass: Output is correct"
else
  echo "Fail: Output is not correct"
  cat DIFFOUT
  rm TESTOUT
  rm DIFFOUT
  exit 1
fi

echo
echo "All tests passed."

rm TESTOUT
rm DIFFOUT
exit 0

