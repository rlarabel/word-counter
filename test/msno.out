MissingNo.: 23
for: 6
and: 35
the: 92
is: 11
an: 12
Pokemon: 29
species: 2
in: 29
video: 5
games: 7
Red: 5
Due: 2
to: 40
of: 41
certain: 2
in-game: 4
players: 10
can: 8
encounter: 7
a: 32
glitch.: 2
graphical: 3
by: 12
increasing: 2
number: 2
items: 2
sixth: 2
player's: 4
inventory: 2
glitch's: 2
guides: 2
game: 12
Nintendo: 5
warned: 3
that: 20
encountering: 3
glitch: 5
may: 4
IGN: 2
MissingNo.'s: 3
Blue: 4
was: 5
one: 3
most: 2
glitches: 4
commented: 2
on: 9
its: 5
role: 2
Fans: 2
have: 6
attempted: 2
incorporate: 2
as: 22
part: 2
games': 2
actual: 2
==: 10
Game: 5
series: 2
with: 11
In: 4
these: 2
player: 10
goal: 2
capture: 3
creatures: 2
Pokemon.: 2
abilities: 2
other: 2
Pokemon,: 3
also: 3
game's: 7
such: 4
travel: 3
The: 5
from: 6
not: 6
are: 2
but: 2
it: 7
events: 2
appear: 2
could: 2
your: 2
or: 4
remove: 2
several: 2
games.: 2
Viridian: 2
City: 2
location.: 2
uses: 2
move: 2
Cinnabar: 3
Island: 2
up: 2
generate: 2
invalid: 2
area: 2
within: 2
data: 4
be: 5
encountered: 2
area.: 2
this: 4
used: 2
tutorial,: 2
character's: 4
name: 4
temporarily: 2
read: 3
same: 2
viewing: 2
being: 3
appears: 2
even: 2
cut: 2
commonly: 2
how: 2
game,: 2
article,: 3
helped: 2
book: 3
MissingNo[.]: 2
his: 3
Wilma: 2
Bainbridge,: 2
then: 2
2019: 2
Ars: 2
at: 2
belief: 2
professor: 2
world: 2
fan: 2
theories: 2
related: 2
Explained: 2
-: 2

1274 msno.txt

